src/ameba/formatter/util.cr

```diff
@@ -4,42 +4,57 @@ module Ameba::Formatter
       message.try &.gsub(/\x1b[^m]*m/, "").presence
     end
 
-    def affected_code(source, location, context_lines = 0, max_length = 100, placeholder = " ...", prompt = "> ")
+    def trim(str, max_length = 120, placeholder = " ...")
+      if str.size > max_length
+        str = str[0, max_length - placeholder.size - 1] + placeholder
+      end
+      str
+    end
+
+    def context(lines, lineno, context_lines = 3)
+      pre_context, post_context = %w[], %w[]
+
+      lines.each_with_index do |line, i|
+        case i + 1
+        when lineno - context_lines...lineno
+          pre_context << trim(line)
+        when lineno + 1..lineno + context_lines
+          post_context << trim(line)
+        end
+      end
+
+      # remove empty lines at the beginning/end
+      pre_context.shift? unless pre_context.first?.presence
+      post_context.pop? unless post_context.last?.presence
```

Comment:
what's the point of using `.presence` here?

Reply:
To check for `blank?` strings.